engine/server/engine/centralized_logs/client_implementations/persistent_volume/file_layout/per_hour_file_layout.go

```diff
@@ -0,0 +1,141 @@
+package file_layout
+
+import (
+	"fmt"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/client_implementations/persistent_volume/logs_clock"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/client_implementations/persistent_volume/volume_consts"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/client_implementations/persistent_volume/volume_filesystem"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/slices"
+	"math"
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	// basepath year/week/day/hour/
+	perHourDirPathFmtStr = "%s%s/%s/%s/%s/"
+
+	// ... enclave-uuid/service-uuid<filetype>
+	perHourFilePathFmtSt = perHourDirPathFmtStr + "%s/%s%s"
+)
+
+type PerHourFileLayout struct {
+	time             logs_clock.LogsClock
+	baseLogsFilePath string
+}
+
+func NewPerHourFileLayout(time logs_clock.LogsClock, baseLogsFilePath string) *PerHourFileLayout {
```

Comment:
This implementation of `FileLayout` stores log files per hour so that logs can be removed per hour.